micro_benchmarking_pytorch.py

```diff
@@ -145,78 +145,78 @@ def rendezvous(distributed_parameters):
     torch.distributed.init_process_group(backend=distributed_parameters['dist_backend'], init_method=distributed_parameters['dist_url'], rank=distributed_parameters['rank'], world_size=distributed_parameters['world_size'])
     print("Rendezvous complete. Created process group...")
 
-def run_benchmarking_wrapper(net, batch_size, iterations, flops_prof_step, amp_opt_level, run_fp16, kineto, autograd_profiler, dataparallel, distributed_dataparallel, device_ids=None, distributed_parameters=None):
-    if (dataparallel or distributed_dataparallel):
-        ngpus = len(device_ids) if device_ids else torch.cuda.device_count()
+def run_benchmarking_wrapper(distributed_parameters=None):
+    if (args.dataparallel or args.distributed_dataparallel):
+        ngpus = len(args.device_ids) if args.device_ids else torch.cuda.device_count()
     else:
         ngpus = 1
 
-    if (distributed_dataparallel):
+    if (args.distributed_dataparallel):
         # Assumption below that each process launched with --distributed_dataparallel has the same number of devices visible/specified
         distributed_parameters['world_size'] = ngpus * distributed_parameters['world_size']
         distributed_parameters['rank'] = ngpus * distributed_parameters['rank']
-        mp.spawn(run_benchmarking, nprocs=ngpus, args=(ngpus, net, batch_size, iterations, flops_prof_step, amp_opt_level, run_fp16, kineto, autograd_profiler, dataparallel, distributed_dataparallel, device_ids, distributed_parameters))
+        mp.spawn(run_benchmarking, nprocs=ngpus, args=(ngpus, args, distributed_parameters))
```

Comment:
Here are some thoughts:
1. We want to keep the code clear for the reader to understand where we are referring to the `args` passed into main vs where we are referring to the args passed to `run_benchmarking`.
2. `mp.spawn` api needs an argument explicitly called `args`

With those in mind, here's my proposal:
1. Don't modify the `args` object coming in from the main function (avoids surprises when debugging and inspecting that `args` object) anywhere *outside main*.
2. Create a deepcopy of the args object for use in internal functions and roll everything into that object. Keeps the interface clean and is extensible.

Reply:
Just pushed a new update addressing this